src/hotspot/os/linux/globals_linux.hpp

```diff
@@ -92,7 +92,9 @@
   develop(bool, DelayThreadStartALot, false,                            \
           "Artificially delay thread starts randomly for testing.")     \
                                                                         \
-
+  product(bool, UseMadvPopulateWrite, false, DIAGNOSTIC,                \
+          "Use MADV_POPULATE_WRITE in os::pd_pretouch_memory.")         \
```

Comment:
I would make this default true. All I intended with my proposal is to have a fallback mechanism if we encounter problems and we want to exclude the madvise as a possible cause. But seeing that the perf gains are real and significant, I would enable it by default.